### PR TITLE
Use include-tagged macro for high level client docs

### PR DESCRIPTION
**Pull request by nik9000**
This should make it simpler to include example snippets from
tests.

Relates to #23351

### CODE REVIEW

**Comment by nik9000** [comment]
Needs https://github.com/elastic/docs/pull/151

File: docs/java-rest/high-level/document/index.asciidoc [hunk 1]
```diff
@@ -1 +1,5 @@
+:doc-tests: {docdir}/../../client/rest-high-level/src/test/java/org/elasticsearch/client/documentation
+
 include::delete.asciidoc[]
+
+:doc-tests!:
```
Comment (dadoonet) [inline docs/java-rest/high-level/document/index.asciidoc]: I'm curious about what is the effect of this?

**Review by dadoonet — APPROVED** [review]
It looks very nice. Thanks for doing that!

File: docs/java-rest/high-level/document/index.asciidoc (hunk 1 above)
Comment (nik9000) [inline docs/java-rest/high-level/document/index.asciidoc]: Clears the variable so it isn't visible outside of this section.

Comment (dadoonet) [inline docs/java-rest/high-level/document/index.asciidoc]: Good to know. Thanks!

**Comment by nik9000** [comment]
Thanks for reviewing @dadoonet! I've just merged the required change to the docs repository so I've merged this too.

**Comment by dadoonet** [comment]
❤️ ! awesome !

**Comment by nik9000** [comment]
> ❤️ ! awesome !

Enjoy!

**Pull request merged**